inc/saiswitch.h

```diff
@@ -3000,6 +3003,17 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_AVAILABLE_SYSTEM_VOQS,
 
+    /**
+     * @brief Default SAI POE ID
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_POE_DEVICE
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
```

Comment:
Shouldn't we have `@condition SAI_SWITCH_ATTR_TYPE == SAI_SWITCH_TYPE_POE` here?
Is it the only switch attribute that is valid for SAI_SWITCH_TYPE_POE ?

Reply:
From the metadata check, I see to add the conditional statement does not allow the type to be an sai_object_id_t. (SAI_OBJECT_TYPE_POE_DEVICE)

(Conditional wants the flags to be: @flags MANDATORY_ON_CREATE | CREATE_ONLY, but this is not allowed with sai_object_id_t. 
SAI_SWITCH_ATTR_DEFAULT_POE_DEVICE_ID: marked as conditional, but invalid creation flags: 0x4
SAI_SWITCH_ATTR_DEFAULT_POE_DEVICE_ID: attribute is create_only and it's an object id, this is not allowed)

Reply:
it cannot be allowed on the switch object, because it make no sense, if you have not switch yet, how you would get other objects ?

Reply:
can you explain what would be the purpose here if you set POE instance here ?

Reply:
The POE instance here will be an entry point for initialization. (Graph dependency)
When creating a POE device object, you are initializing the software and hardware components that are needed for POE to run.